blank/webpack.config.js

```diff
@@ -99,192 +120,188 @@ const stylesLoaderConfig = {
+      new webpack.DefinePlugin({
+        IS_BROWSER: true,
+        HUBTYPE_API_URL: JSON.stringify(process.env.HUBTYPE_API_URL),
+        WEBCHAT_PUSHER_KEY: JSON.stringify(process.env.WEBCHAT_PUSHER_KEY),
       }),
     ],
   }
 }
 
 function botonicWebviewsConfig(mode) {
   return {
-    optimization: {
-      minimizer: [terserPlugin],
-    },
+    optimization: optimizationConfig,
     mode: mode,
     devtool: sourceMap(mode),
     target: 'web',
-    entry: path.resolve('webpack-entries', 'webviews-entry.js'),
+    entry: path.resolve(WEBPACK_ENTRIES.DIR, WEBPACK_ENTRIES.WEBVIEWS),
     output: {
-      path: path.resolve(__dirname, 'dist/webviews'),
       filename: 'webviews.js',
       library: 'BotonicWebview',
-      libraryTarget: 'umd',
+      libraryTarget: UMD_LIBRARY_TARGET,
       libraryExport: 'app',
+      path: path.resolve(OUTPUT_PATH, 'webviews'),
     },
     module: {
       rules: [
         babelLoaderConfig,
-        {
-          test: /\.(png|svg|jpg|gif)$/,
-          use: [
-            {
-              loader: 'file-loader',
-              options: {
-                outputPath: '../assets',
-              },
-            },
-          ],
-        },
+        fileLoaderConfig('../assets'),
         stylesLoaderConfig,
       ],
     },
     resolve: resolveConfig,
     plugins: [
       new HtmlWebpackPlugin({
-        template: path.resolve(botonicPath, 'src', 'webview.template.html'),
+        template: path.resolve(BOTONIC_PATH, 'src', TEMPLATES.WEBVIEWS),
         filename: 'index.html',
       }),
       imageminPlugin,
-      new webpack.EnvironmentPlugin({
-        HUBTYPE_API_URL: null,
-        BOTONIC_TARGET: 'webviews',
+      new webpack.DefinePlugin({
+        IS_BROWSER: true,
+        HUBTYPE_API_URL: JSON.stringify(process.env.HUBTYPE_API_URL),
       }),
     ],
   }
 }
 
-function botonicServerConfig(mode) {
+function botonicNodeConfig(mode) {
   return {
-    optimization: {
-      minimizer: [terserPlugin],
-    },
-    context: root,
+    context: ROOT,
+    optimization: optimizationConfig,
     mode: mode,
     devtool: sourceMap(mode),
     target: 'node',
-    entry: path.resolve('webpack-entries', 'node-entry.js'),
+    entry: path.resolve(WEBPACK_ENTRIES.DIR, WEBPACK_ENTRIES.NODE),
+    resolve: resolveConfig,
     output: {
       filename: 'bot.js',
       library: 'bot',
-      libraryTarget: 'umd',
+      libraryTarget: UMD_LIBRARY_TARGET,
       libraryExport: 'app',
+      path: OUTPUT_PATH,
     },
     module: {
-      rules: [babelLoaderConfig, fileLoaderConfig, nullLoaderConfig],
+      rules: [babelLoaderConfig, fileLoaderConfig('assets'), nullLoaderConfig],
     },
-    resolve: resolveConfig,
     plugins: [
       new CleanWebpackPlugin({ cleanOnceBeforeBuildPatterns: ['dist'] }),
       imageminPlugin,
-      new webpack.EnvironmentPlugin({
-        HUBTYPE_API_URL: null,
-        BOTONIC_TARGET: 'node',
+      new webpack.DefinePlugin({
+        IS_NODE: true,
```

Comment:
Why not specify `IS_NODE` and `IS_BROWSER` in each configuration?

Reply:
yes, it will be more explicit, added them